src/main/g8/src/main/scala/com/example/WebServer.scala

```diff
@@ -0,0 +1,34 @@
+package com.example
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.stream.ActorMaterializer
+import scala.io.StdIn
+
+object WebServer {
+  def main(args: Array[String]) {
+
+    implicit val system = ActorSystem("my-system")
+    implicit val materializer = ActorMaterializer()
+    // needed for the future flatMap/onComplete in the end
+    implicit val executionContext = system.dispatcher
+
+    val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+
+    println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
+    StdIn.readLine() // let it run until user presses return
```

Comment:
ah, it's probably wrong here. it should perform the unbind and actor system shutdown after ENTER. Here is another example: https://github.com/akka/akka/blob/master/akka-docs-new/src/test/scala/quickstart/HelloWorldApp.scala#L18